gutenberg.php

```diff
@@ -105,6 +105,23 @@ function gutenberg_menu() {
 }
 add_action( 'admin_menu', 'gutenberg_menu', 9 );
 
+/**
+ * Modify WP admin bar.
+ *
+ * @param WP_Admin_Bar $wp_admin_bar Core class used to implement the Toolbar API.
+ */
+function modify_admin_bar( $wp_admin_bar ) {
+	if ( gutenberg_use_widgets_block_editor() ) {
+		$wp_admin_bar->add_menu(
+			array(
+				'id'   => 'widgets',
+				'href' => admin_url( 'themes.php?page=gutenberg-widgets' ),
```

Comment:
This link is in two places now, maybe let's use a constant or a variable instead of hardcoding it? It would be easier to change if needed and we would also reduce the risk of making a typo.

Reply:
Sure, I'm not familiar with PHP though, any suggestion on where to put this constant?

Reply:
I think at the top of gutenberg.php should be fine

Reply:
Is 265ddc1 what you meant?

Reply:
I think I disagree with @adamziel 😇

The problem is that, because WordPress doesn't use namespaces, every top level variable, function, and constant we add becomes, for better or worse, a part of the public API that we have to maintain backwards compatibility for.

I don't think the repetition is severe enough in this case to warrant that kind of a commitment.

---

Having said that, if we still want to use a constant, let's avoid `define()` which is, in WordPress, used for configuration constants. 

Instead let's use the `const` keyword which should support arbitrary expressions as of PHP 5.6.

5.6 is the minimum PHP version that WordPress supports.

We can also mark it `@access private` so that we're not exposing this as part of our public API.

```php
/**
 * @access priavte
 */
const GUTENBERG_WIDGETS_LINK = admin_url( 'themes.php?page=gutenberg-widgets' ) );
```

https://www.php.net/const

Reply:
Wow TIL! `define` is just the very first result from my Google search 😆 

Do we want to go with `const` keyword and `@access private`? Or just inline it everywhere? I don't have a strong preference of this 🤔 .

Reply:
> Like any other PHP static variable, static properties may only be initialized using a literal or constant before PHP 5.6; expressions are not allowed. In PHP 5.6 and later, the same rules apply as const expressions: some limited expressions are possible, provided they can be evaluated at compile time.

Hmmm, seems like we can't use `admin_url()` in `const` expressions.

Reply:
I think just keep the duplication. It's very common 🙂

https://github.com/WordPress/wordpress-develop/search?q=admin_url

Reply:
@kevin940726 I tend to agree with @noisysocks and just use the admin_url call with the value in multiple places. It's cheaper than defining a constant which people will use for whatever reason. Plus it does not solve @adamziel 's main problem which is the hardcoding part. Unfortunately there is no place to bring that route from `'themes.php?page=gutenberg-widgets'` as it's a made up, more or less, route.

```
$wp_admin_bar->add_menu(
			array(
				'id'   => 'widgets',
				'href' => admin_url( 'themes.php?page=gutenberg-widgets' ),
...
```

This is better imo, no `define` no `const`.

Reply:
Done! Reverted back to the old commit.